### ISSUE TITLE
rule annotations for wasm entrypoints

### ISSUE THREAD

**Issue by srenatus** [open-policy-agent/opa#3459]
It would be nice if I could annotate a rule in a way such that `opa build -t wasm a.rego b.rego c.rego` would pick up the desired entrypoints and use them for building a wasm bundle.

Right now, any entrypoint must be provided by `-e a/b/c` (for `data.a.b.c`), and that's a little cumbersome when dealing with multiple of them.

👉 #3422 could benefit from that, with many rego files and many entrypoints.

**Comment by tsandall** [comment]
Marking entrypoints in the policy would also be useful for optimization like partial eval that require a path today.

**Comment by srenatus** [comment]
Misclicked sorry

**Comment by itaysk** [comment]
This would also be useful for documentation/integration purposes, as suggested in https://github.com/open-policy-agent/opa/issues/2180#issuecomment-778739067 and explained in https://hackmd.io/k6k7TVkFQ2m-ElQ2OT7nBQ#Entrypoints
TL;DR: I'm currently integrating some existing policies which was written by different people/projects, and it's useful to just know what am I supposed to evaluate there, in a machine readable way.

I know that the motivation here is entirely different, but perhaps we could win-win with this one.

**Comment by anderseknert** [comment]
Another benefit of this could be that e.g. strict mode, or future linters, could report on unused variables and rules even at the top level, if they aren't referenced by any of the entrypoint rules.

**Comment by philipaconrad** [comment]
The two commands that are going to be most affected by this are the `build` and `eval` commands, since they're the only ones that support providing entrypoints explicitly. `inspect` *might* need some extension to print out which rules have entrypoint annotations.
